### PR TITLE
fix(ui): backport memory leak

### PR DESCRIPTION
**Pull request by danielhalupka**
Refs: MON-2886

### CODE REVIEW

File: www/include/home/customViews/index.ihtml [hunk 1]
```diff
@@ -852,8 +866,38 @@ jQuery(function () {
...
+        var defaultPosition = getTabPos(data.current);
+
         jQuery("#tabs").tabs({
-          active: getTabPos(data.current)
+          active: defaultPosition,
+          beforeLoad: function( event, ui ) {
+            // prevent memory overflow
+            if (tabLoops > 10 && $(event.target).tabs('option', 'active') == defaultPosition) {
+                if(window.parent){
+                  console.log('here')
```
Comment (kduret) [inline www/include/home/customViews/index.ihtml]: remove debug

File: www/include/home/customViews/index.ihtml [hunk 2]
```diff
@@ -852,8 +866,38 @@ jQuery(function () {
...
+                  window.parent.location.reload();
+                }else{
+                  console.log('here2')
```
Comment (kduret) [inline www/include/home/customViews/index.ihtml]: remove debug

File: www/include/home/customViews/index.ihtml [hunk 3]
```diff
@@ -253,9 +253,23 @@
 <script>
 {literal}
 
+(function($) {
+                $.fn.purgeFrame = function() {
```
Comment (kduret) [inline www/include/home/customViews/index.ihtml]: bad indentation on this block

File: www/include/home/customViews/views.php [hunk 4]
```diff
@@ -110,206 +111,214 @@
...
+                        custom_view_id: viewId,
+                        widget_id: widgetId
+                    },
+                    success: function (response) {
+                        var view = response.getElementsByTagName('custom_view_id');
+                        var error = response.getElementsByTagName('error');
+                        if (typeof(view) != 'undefined') {
+                            var viewId = view.item(0).firstChild.data;
+                            jQuery('#tabs').tabs('load', getTabPos(viewId));
+                            //window.top.location = './main.php?p=103&currentView='+viewId;
```
Comment (victorvassilev) [inline www/include/home/customViews/views.php]: remove comment if not needed

**Comment by danielhalupka** [comment]
Requested changes are done

File: www/include/home/customViews/views.php [hunk 5]
```diff
@@ -110,206 +111,213 @@
...
+            <?php 
+                $i = 0;
+                foreach ($widgets as $widgetId => $widget) { ?>
+                    createFrame('<?php echo $widget['url']; ?>?widgetId=' +<?php echo $widgetId;?> +'&customViewId=' + viewId,$('#widget_cont_<?php echo $widgetId;?>'),"widget_" + viewId + "_<?php echo $widgetId;?>", <?php echo $i;?>);
+            <?php $i++; } ?>
+        }
 
-    jQuery(".ui-icon-wrench").each(function(index, element) {
+        jQuery(".ui-icon-wrench").each(function(index, element) {
                                         var tmp = jQuery(element).parents('.portlet').attr('name')
```
Comment (sc979) [inline www/include/home/customViews/views.php]: please correct these lines indentation

File: www/include/home/customViews/views.php [hunk 6]
```diff
@@ -110,206 +111,213 @@
...
-                                } else if (typeof(error) != 'undefined') {
-                                    var errorMsg = error.item(0).firstChild.data;
-                                }
-                            }
-            });
-        }
-    });
-}
+    function createFrame(url, parent,name, idx) {
+                var $frame = $('<iframe/>')
```
Comment (sc979) [inline www/include/home/customViews/views.php]: odd indentation

Comment (victorvassilev) [reply www/include/home/customViews/views.php]: applied

File: www/include/home/customViews/views.php (hunk 5 above)
Comment (victorvassilev) [reply www/include/home/customViews/views.php]: applied

File: www/include/home/customViews/views.php [hunk 7]
```diff
@@ -110,206 +111,213 @@
...
-    }
+            jQuery(".portlet-header .show-hide").click(function () {
+                jQuery(this).toggleClass("ui-icon-minusthick").toggleClass("ui-icon-plusthick");
+                jQuery(this).parents(".portlet:first").find(".portlet-content").toggle();
+            });
+            
+            <?php 
+                $i = 0;
+                foreach ($widgets as $widgetId => $widget) { ?>
+                    createFrame('<?php echo $widget['url']; ?>?widgetId=' +<?php echo $widgetId;?> +'&customViewId=' + viewId,$('#widget_cont_<?php echo $widgetId;?>'),"widget_" + viewId + "_<?php echo $widgetId;?>", <?php echo $i;?>);
```
[the remark this answers is not in the document]
Comment (victorvassilev) [reply www/include/home/customViews/views.php]: applied

File: www/include/home/customViews/views.php [hunk 8]
```diff
@@ -110,206 +111,213 @@
...
-var permission = <?php echo ($permission === true) ? 1 : 0; ?>;
-var ownership = <?php echo ($ownership === true) ? 1 : 0; ?>;
-var wrenchSpan = '<span class="ui-icon ui-icon-wrench"></span>';
-var trashSpan = '<span class="ui-icon ui-icon-trash"></span>';
-var idUser = "<?php echo $centreon->user->user_id;?>";
+    var columnClass = "<?php echo $columnClass;?>";
+    var viewId = "<?php echo $viewId;?>";
+    var deleteWdgtMessage = "<?php echo _("Deleting this widget might impact users with whom you are sharing this view. Are you sure you want to do it?");?>";
+    var deleteViewMessage = "<?php echo _("Deleting this view might impact other users. Are you sure you want to do it?");?>";
+    var setDefaultMessage = "<?php echo _("Set this view as your default view?");?>";
```
[the remark this answers is not in the document]
Comment (victorvassilev) [reply www/include/home/customViews/views.php]: thanks, done

File: www/include/home/customViews/views.php [hunk 9]
```diff
@@ -39,6 +39,7 @@
 require_once _CENTREON_PATH_ . 'www/class/centreonCustomView.class.php';
 require_once _CENTREON_PATH_ . 'www/class/centreonWidget.class.php';
 require_once _CENTREON_PATH_ . 'www/class/centreonDB.class.php';
```
[the remark this answers is not in the document]
Comment (victorvassilev) [reply www/include/home/customViews/views.php]: thanks, cleaned

File: www/include/home/customViews/index.ihtml [hunk 10]
```diff
@@ -852,8 +866,36 @@ jQuery(function () {
...
+                  window.parent.location.reload();
+                }else{
+                  window.location.reload();
+                } 
+            }
+          },
+          activate: function( event, ui ) {
+            for(var i=0;i<window.iframes.length;i++){
+	            window.iframes[i].purgeFrame()
+            }
```
[the remark this answers is not in the document]
Comment (victorvassilev) [reply www/include/home/customViews/index.ihtml]: thanks, fixed

**Pull request merged**